dali/operators/generic/expand_dims.cc

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include <vector>
+#include <set>
+
+#include "dali/core/math_util.h"
+#include "dali/core/static_switch.h"
+#include "dali/core/tensor_layout.h"
+#include "dali/core/tensor_shape_print.h"
+#include "dali/operators/generic/expand_dims.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+DALI_SCHEMA(ExpandDims)
+  .DocStr(R"code(Insert new dimension[s] of extent 1 and inserts new entries in "
+    "the layout (new_axis_names) at these indices in the layout.)code")
```

Comment:
We probably want a shorter first description for the table of operators.
I'd say
```suggestion
  .DocStr(R"code(Insert new dimension(s) with extent 1 to the data shape.

The new dimensions are inserted at the positions specified by ``axes``.
  
If ``new_axis_names`` is provided, the new dimension names will be inserted in the data layout, at the positions specified by ``axes``. If ``new_axis_names`` is not provided, the output data layout will be empty.")code")
```